vendor/plugins/edu.harvard.dce.paella.heartbeatSender/heartbeat_sender.js

```diff
@@ -0,0 +1,68 @@
+// A version of the deprecated es.upv.paella.UserTrackingCollectorPlugIn, 
+// shaved down to just send the heartbeat.
+
+var classDef = {
+  heartbeatTimer:null,
+
+  getName: function() {
+    return "edu.harvard.dce.paella.heartbeatSender";
+  },
+
+  setup: function() {
+    var thisClass = this;
+  
+    if (this.config.heartBeatTime > 0) {
+      this.heartbeatTimer = new base.Timer(
+        registerHeartbeat, this.config.heartBeatTime
+      );
+      this.heartbeatTimer.repeat = true;
+    }
+
+    function registerHeartbeat(timer) {
+      var xhr = new XMLHttpRequest();
+      xhr.open('GET', getHeartbeatURL());
+      xhr.send();
+    }
+
+    function getHeartbeatURL() {
+      var videoCurrentTime = parseInt(
+        paella.player.videoContainer.currentTime() +
+        paella.player.videoContainer.trimStart(),
+        10
+      );
+
+      var url = 'https://';
+      url += location.host + '/';
+      url += 'usertracking/?';
+      url += queryStringFromDict({
+        _method: 'PUT',
+        id: paella.player.videoIdentifier,
+        type: 'HEARTBEAT',
+        in: videoCurrentTime,
```

Comment:
OK, I think I know where the confusion is coming from.

https://mathiasbynens.be/notes/javascript-properties

```Note: ECMAScript 3 didn’t allow the use of unquoted reserved words as property names. Here’s the full list of ES3 reserved words: abstract, boolean, break, byte, case, catch, char, class, const, continue, debugger, default, delete, do, double, else, enum, export, extends, false, final, finally, float, for, function, goto, if, implements, import, in, instanceof, int, interface, long, native, new, null, package, private, protected, public, return, short, static, super, switch, synchronized, this, throw, throws, transient, true, try, typeof, var, void, volatile, while, and with. Avoid using these as unquoted property names if backwards compatibility is a concern.```

ES3 is pretty old. I think even IE8 supports ES5.